File: frontend/src/api/knowledge.ts
import { apiClient } from './client'

export type KnowledgeImportRequest = {
  source_type?: string
  title: string
  version?: string | null
  source_url?: string | null
  raw_text: string
}

export type KnowledgeImportResponse = { document_id: string; chunk_count: number }

export type KnowledgeSearchItem = {
  chunk_id: string
  document_id: string
  title: string
  version: string | null
  source_url: string | null
  chunk_index: number
  score: number
  text: string
}

export type KnowledgeSearchResponse = { items: KnowledgeSearchItem[] }

export async function apiKnowledgeImport(payload: KnowledgeImportRequest): Promise<KnowledgeImportResponse> {
  const { data } = await apiClient.post('/api/knowledge/import', payload)
  return data
}

export async function apiKnowledgeSearch(q: string, topK = 5): Promise<KnowledgeSearchResponse> {
  const { data } = await apiClient.get('/api/knowledge/search', { params: { q, top_k: topK } })
  return data
}

File: backend/app/schemas/knowledge.py
from __future__ import annotations

import uuid

from pydantic import BaseModel, Field


class KnowledgeImportRequest(BaseModel):
    source_type: str = Field(default="raw_text", max_length=32)
    title: str = Field(min_length=1, max_length=255)
    version: str | None = Field(default=None, max_length=64)
    source_url: str | None = None
    raw_text: str = Field(min_length=1, max_length=2_000_000)


class KnowledgeImportResponse(BaseModel):
    document_id: uuid.UUID
    chunk_count: int


class KnowledgeSearchItem(BaseModel):
    chunk_id: uuid.UUID
    document_id: uuid.UUID
    title: str
    version: str | None
    source_url: str | None
    chunk_index: int
    score: float
    text: str


class KnowledgeSearchResponse(BaseModel):
    items: list[KnowledgeSearchItem]

